python/tvm/auto_scheduler/task_scheduler.py

```diff
@@ -278,6 +281,47 @@ def __init__(
                 self.group_task_ids.append([])
             self.group_task_ids[self.tag_to_group_id[tag]].append(i)
 
+   
+    def calc_ref_tensors(self):
+
+        try:
+            # calc reference tensors
+            local_builder = LocalBuilder()
+            local_runner  = LocalRunner()
+
+
+            for idx, task in enumerate(self.tasks):
+                print("========== Task %d  (workload key: %s) ==========" %
+                    (idx, task.workload_key))
+                print(task.compute_dag)
+                
+                state = task.compute_dag.get_init_state()
+                original_target = task.target
+                original_target_host = task.target_host
+
+                ref_target = tvm.target.Target("llvm", host="llvm")
+                _ffi_api.SetTarget(task, ref_target, None)
+                measure_inputs = [MeasureInput(task, state)]
+
+                print("Collecting reference output tensors:")
+                # build in silent mode
+                build_results = local_builder.build(measure_inputs, verbose=0)
+
+                results = local_runner.get_ouput(measure_inputs, build_results)
+
+                _ffi_api.SetReferenceTensors(task, results)
```

Comment:
Using of function `_ffi_api.SetReferenceTensors` is a workaround. I'm personally sure that ffi infrastructure allow to set and get values of native object without special setting methods.